Zend/tests/gh7900.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+GH-7900: Arrow function with never return type compile-time errors
+--INI--
+zend.assertions=1
+assert.exception=1
+--FILE--
+<?php
+
+$x = fn(): never => throw new \Exception('Here');
+
+try {
+    var_dump($x());
+} catch (\Exception $e) {
+    echo $e->getMessage(), "\n";
+}
+
+try {
+    assert((fn(): never => 42) && false);
+} catch (\Error $e) {
+    echo $e->getMessage(), "\n";
+}
+
+?>
+--EXPECT--
+Here
+assert(fn(): never => 42 && false)
```

Comment:
The printed AST (which dropped parentheses around the arrow function) looks wrong w.r.t. operator precedence (`&&` binds tighter than `=>`)

Reply:
Parentheses are lost after the parsing phase. This is a longstanding but probably insignificant problem since we only use the AST printing for assert.

Reply:
Oh, that's already the case: https://3v4l.org/j1XK1 :/ needs a new bug report?

Reply:
Our messages crossed :s ACK